src/casync-http.c

```diff
@@ -30,6 +31,40 @@ typedef enum ProcessUntil {
         PROCESS_UNTIL_FINISHED,
 } ProcessUntil;
 
+static CURLcode robust_curl_easy_perform(CURL *curl) {
+        useconds_t sleep_time = 100000;
+        uint trial = 1;
+        uint limit = 10;
+        CURLcode c;
+
+        assert(curl);
+
+        while (trial < limit) {
+
+                c = curl_easy_perform(curl);
+
+                switch (c) {
+
+                case CURLE_COULDNT_CONNECT:
+                        /* Although this is not considered as a transient error by curl,
+                         * this error can happen momentarily while casync is retrieving
+                         * all the chunks from a remote. In this case we want to give
+                         * a break to the server and retry.
+                         */
+                        usleep(sleep_time);
+                        trial++;
+                        sleep_time *= trial;
+                        break;
+
+                default:
+                        return c;
```

Comment:
You can remove this 'return c;' here, the next 'break' is enough.

Reply:
Actually no, the break just take us out of the `switch`, but we also want to escape the `while` loop here.